### PR TITLE
Feature/DTGB-419: Timeline

### PR DESCRIPTION
**Pull request by meirege**
## Description
Timeline desktop + mobile.

## Screenshots (if appropriate):
screencapture-localhost-3000-components-preview-timeline-2018-05-31-10_43_59
screencapture-localhost-3000-components-preview-timeline-2018-05-31-11_16_12

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the style guide CHANGELOG accordingly.
- [ ] I have updated the gent_base CHANGELOG accordingly.

### CODE REVIEW

File: styleguide/components/31-molecules/video/video.twig [hunk 1]
```diff
@@ -8,10 +8,12 @@
             allow="autoplay; encrypted-media"
             allowfullscreen></iframe>
   </div>
-{% if audio_description_src %}
-  {% include '@link' with {
-    'link': audio_description_src,
-    'text': 'Watch this video with audio description'
-  } %}
-{% endif %}
+  <div class="audio-description">
```
Comment (meirege) [inline styleguide/components/31-molecules/video/video.twig]: @delrueba So this fix was necessary to get the damn dots for the timeline video.

File: styleguide/components/41-organisms/feedback-form/feedback-form.twig [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div class="feedback-form {{ modifier }}">
-  <h2><button>{{ 'Can we improve something about this page?' }}<i class="icon-chevron-down" aria-hidden="true" aria-expanded="true" aria-controls="feedback-form-content"></i></button></h2>
+  <h2><button><i class="icon-arrow-down"></i>{{ 'Can we improve something about this page?' }}<i class="icon-chevron-down" aria-hidden="true" aria-expanded="true" aria-controls="feedback-form-content"></i></button></h2>
```
Comment (delrueba) [inline styleguide/components/41-organisms/feedback-form/feedback-form.twig]: Why the first icon? Should be aria-hidden too.

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 3]
```diff
@@ -0,0 +1,397 @@
+dl.timeline {
+  width: 100%;
+
+  * {
+    font-size: .8rem;
+   }
+
+  .timeline-slot {
+    @include theme('background-color', 'color-secondary', 'timeline-slot-default-background');
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: Just keep it transparent. I don't see any indication in design that this has a background color.

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+
+    display: flex;
+    position: relative;
+    flex-wrap: wrap;
+    width: 100%;
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: Obsolete, div's are block elements.

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+      &::before {
+        position: absolute;
+        left: -1.6rem;
+        font-size: 1.6em;
+      }
+    }
+
+    &-date {
+      display: block;
+      width: 100%;
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss:119]: obsolete, display block is 100%

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+      .responsive-video::before,
+      .audio-description::after {
+        display: none;
+      }
+    }
+
+    // Timeline-slot title styling.
+    &-title {
+      &[aria-expanded="true"] {
+        @include icon('minus');
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: color is not correct

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+      }
+
+      &[aria-expanded="false"] {
+        @include icon('plus');
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: color is not correct

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+      }
+    }
+
+    > dd {
+      @include theme('border-color', 'color-primary', 'timeline-line-color');
+
+      display: flex;
+      position: relative;
+      flex-wrap: wrap;
+      width: 100%;
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss:165]: Obsolete, flex is block.

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+      }
+
+      &.active::before {
+        @include theme('background-color', 'color-primary', 'timeline-slot-active-indicator-background');
+      }
+
+      // Video component styling inside a timeline-slot.
+      .video-wrapper {
+        position: relative;
+        width: calc(100% + 4.6rem + 4px);
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: What is the origin of this 4px? It's unclear.

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+              @include tablet {
+                left: calc(5rem + 1px);
+              }
+            }
+          }
+        }
+
+        .responsive-video {
+          // Dots before a video intersection.
+          &::before {
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss:262]: Combine this in one query with the other dot.
Better for maintenance.

**Review by delrueba — CHANGES_REQUESTED** [review]

File: styleguide/components/41-organisms/timeline/timeline.twig [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+                "alt": "placeholder image alternative text",
+                "caption": "image caption"
+              }
+            ]
+          } %}
+        </div>
+      </div>
+    </dd>
+    <dd>
+    <button class="timeline-slot-title" aria-expanded="true" aria-controls="timeline-slot-content"><h3>Title</h3></button>
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/timeline.twig]: missing wrapper

File: styleguide/components/41-organisms/timeline/timeline.twig [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+      <span class="timeline-slot-date">7 november 2017</span>
+    </dd>
+    <dd>
+      <button class="timeline-slot-title" aria-expanded="true" aria-controls="timeline-slot-content"><h3>Cupcake ipsum lolly cake marzipan</h3></button>
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/timeline.twig]: missing wrapper

File: styleguide/components/41-organisms/timeline/_timeline.scss [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+      &[aria-expanded="false"] {
+        @include icon('plus');
+      }
+
+      position: relative;
+      margin-left: -1.6rem;
+      padding-left: 1.6rem;
+      border: 0;
+      background: transparent;
+
```
Comment (delrueba) [inline styleguide/components/41-organisms/timeline/_timeline.scss]: set with to 100% (or 100% + 1.6rem) to have a bigger touch surface

**Review by delrueba — CHANGES_REQUESTED** [review]

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 3 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: This has been done for subthemes... If we wish to use 2 other colors, this won't break anything.

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 4 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: 👍

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 6 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: ?

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 7 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: ?

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 13 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: Width of the title?

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 5 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss:119]: This is necessary or the title and date div / span will be next to each other...

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 8 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss:165]: 👍

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 9 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: True, It's from the timeline border. I'll change this.

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 10 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/_timeline.scss:262]: 👍

File: styleguide/components/41-organisms/timeline/timeline.twig (hunk 12 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/timeline.twig]: Fixed.

File: styleguide/components/41-organisms/timeline/timeline.twig (hunk 11 above)
Comment (meirege) [reply styleguide/components/41-organisms/timeline/timeline.twig]: Fixed.

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 13 above)
Comment (delrueba) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: Or the button

File: styleguide/components/31-molecules/video/video.twig (hunk 1 above)
Comment (meirege) [reply styleguide/components/31-molecules/video/video.twig]: 👍

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 5 above)
Comment (delrueba) [reply styleguide/components/41-organisms/timeline/_timeline.scss:119]: No, that was caused by the template error.

File: styleguide/components/41-organisms/timeline/_timeline.scss (hunk 6 above)
Comment (delrueba) [reply styleguide/components/41-organisms/timeline/_timeline.scss]: inherits from button, it's light gray

**Review by delrueba — APPROVED** [review]

**Pull request merged**